client/client.go

```diff
@@ -201,13 +79,19 @@ func (c *Client) DoRequest(url, httpMethod string, resourceData interface{}) ([]
 			return nil, fmt.Errorf("unable to create request; err: %v", err)
 		}
 	}
+
 	req.Header.Add("content-type", "application/json")
-	// The TokenType returned by getKeycloakToken is "bearer", but if we use it here we
-	// will get error "Failed to get roles: tokenstring should not contain 'bearer '\n".
-	// If we change it to "Bearer" it works normally.
-	// See: https://cyralinc.atlassian.net/browse/ENG-4408
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.Token))
-	//req.Header.Add("Authorization", fmt.Sprintf("%s %s", c.TokenType, c.Token))
+	if c.TokenSource != nil {
+		if token, err := c.TokenSource.Token(); err != nil {
+			return nil, fmt.Errorf("unable to retrieve authorization token. error: %v", err)
+		} else {
+			log.Printf("[DEBUG] Token Type: %s", token.Type())
+			log.Printf("[DEBUG] Access Token: %s", token.AccessToken)
```

Comment:
We should redact the access token before printing. Please also make sure we also redact the client secret if we print it. This was an old mistake we should improve moving forward.